type-overrides/src/lib.rs

```diff
@@ -0,0 +1,48 @@
+///
+/// This lib contains both standard imports and imports shuttle.
+/// Shuttle is a Rust crate that facilitates multithreaded testing. It has its own scheduler
+/// and can efficiently detect bugs in concurrent code. The downside is that we need to replace
+/// all imports by those from Shuttle.
+///
+/// Instead of importing from std, rand, and so on, import the following from solana-type-override,
+/// and include the 'shuttle-test' feature in your crate to use shuttle.
+pub mod executor {
+    #[cfg(not(feature = "shuttle-test"))]
+    pub use futures::executor::*;
+    #[cfg(feature = "shuttle-test")]
+    pub use shuttle::future::*;
+}
+
+pub mod hint {
+    #[cfg(feature = "shuttle-test")]
+    pub use shuttle::hint::*;
+    #[cfg(not(feature = "shuttle-test"))]
+    pub use std::hint::*;
+}
+
+pub mod lazy_static {
+    #[cfg(not(feature = "shuttle-test"))]
+    pub use lazy_static::*;
+    #[cfg(feature = "shuttle-test")]
+    pub use shuttle::lazy_static::*;
+}
+
+pub mod rand {
+    pub use rand::*;
+    #[cfg(feature = "shuttle-test")]
+    pub use shuttle::rand::{thread_rng, Rng, RngCore};
```

Comment:
Why not `*`? :P

Reply:
Because rustc complains about it:

```
warning: ambiguous glob re-exports
  --> type-overrides/src/lib.rs:31:13
   |
31 |     pub use rand::*;
   |             ^^^^^^^ the name `rngs` in the type namespace is first re-exported here
32 |     #[cfg(feature = "shuttle-test")]
33 |     pub use shuttle::rand::*;
   |             ---------------- but the name `rngs` in the type namespace is also re-exported here
   |
   = note: `#[warn(ambiguous_glob_reexports)]` on by default

warning: ambiguous glob re-exports
  --> type-overrides/src/lib.rs:31:13
   |
31 |     pub use rand::*;
   |             ^^^^^^^ the name `thread_rng` in the value namespace is first re-exported here
32 |     #[cfg(feature = "shuttle-test")]
33 |     pub use shuttle::rand::*;
   |             ---------------- but the name `thread_rng` in the value namespace is also re-exported here

warning: unused import: `shuttle::rand::*`
  --> type-overrides/src/lib.rs:33:13
   |
33 |     pub use shuttle::rand::*;
   |             ^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

warning: `solana-type-overrides` (lib) generated 3 warnings (run `cargo fix --lib -p solana-type-overrides` to apply 1 suggestion)
error[E0432]: unresolved import `solana_type_overrides::rand::thread_rng`
  --> program-runtime/src/loaded_programs.rs:23:16
   |
23 |         rand::{thread_rng, Rng},
   |                ^^^^^^^^^^ no `thread_rng` in `rand`

warning: unused import: `Rng`
  --> program-runtime/src/loaded_programs.rs:23:28
   |
23 |         rand::{thread_rng, Rng},
   |                            ^^^
   |
   = note: `#[warn(unused_imports)]` on by default
```

Reply:
`#[allow(ambiguous_glob_reexports)]`?

Reply:
It makes the warning go away, but it doesn't solve the compilation error (see the bottom of my previous comment):

```
error[E0432]: unresolved import `solana_type_overrides::rand::thread_rng`
  --> program-runtime/src/loaded_programs.rs:23:16
   |
23 |         rand::{thread_rng, Rng},
   |                ^^^^^^^^^^ no `thread_rng` in `rand`

```

Reply:
I don't get this error locally, but I get 

```
warning: unused import: `shuttle::rand::*`
  --> type-overrides/src/lib.rs:34:13
   |
34 |     pub use shuttle::rand::*;
   |             ^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default
```

It looks like you can't have two glob imports, but the 2nd one must explicitly override

Reply:
If I try to build the SVM with the shuttle feature, I get the aforementioned error.
`cargo build --manifest-path svm/Cargo.toml --features="shuttle-test"`

Building everything with the shuttle feature won't work because I haven't patched all the crates with the new types.